lib/adapters/parsers/csv.js

```diff
@@ -33,7 +33,7 @@ class CSVParser extends base {
                     self.doneParsing = true;
                     emit(buffer);
                     buffer = [];
-                    csvStream.end();
+                    csvStream.emit('end');
```

Comment:
Does `end()` not emit `'end'`? Do you not have to call `end()` at some point to clean the stream up properly?